libr/cons/dietline.c

```diff
@@ -774,8 +789,9 @@ R_API const char *r_line_readline_cb_win(RLineReadCallback cb, void *user) {
 			}
 			break;
 		case 39:// right arrow
-			I.buffer.index = I.buffer.index < I.buffer.length?
-					 I.buffer.index + 1: I.buffer.length;
+			I.buffer.index = I.buffer.index < I.buffer.length
```

Comment:
this check is probably not enough because its not considering the charsize. also the utf8_charsize should get the length to not override the buffer